modelon/impact/client/entities/workspace.py

```diff
@@ -1126,3 +1210,53 @@ def from_conversion_operation(
     ) -> Workspace:
         assert isinstance(operation, WorkspaceConversionOperation)
         return cls(**kwargs, service=operation._sal)
+
+    @property
+    def received_from(self) -> ReceivedFromWorkspace:
+        """Returns the ReceivedFromWorkspace class."""
+        return ReceivedFromWorkspace(self)
+
+    def get_published_workspace(
+        self, model_name: Optional[str] = None
+    ) -> Optional[PublishedWorkspace]:
+        """Returns the uploaded published workspace.
+
+        Args:
+            model_name: Class path of the modelica model. If specified,
+             an app mode published workspace corresponding to the local workspace
+             is returned(if it exists).
+
+        Returns:
+            An PublishedWorkspace class object.
+
+        Example::
+
+            published_ws = workspace.get_published_workspace()
+
+        """
+        recieved_from = self.definition.received_from
+        if recieved_from:
+            raise RemotePublishedWorkspaceLinkError(
+                "The workspace is linked to a published workspace with ID:"
+                f"{recieved_from.sharing_id}. Please rename the workspace to unlink"
+                "it from the published workspace, if you wish to publish it!"
```

Comment:
"This workspace is imported from published workspace with ID:"
 f"{recieved_from.sharing_id}. Use received_from.get_workspace() to retrieve the information" 

Do we have workspace rename functionality in client? Anyway, I don't think we need to talk about that here.

Reply:
workspace.rename('renamed workspace')